packages/jest-haste-map/src/index.js

```diff
@@ -325,6 +325,28 @@ class HasteMap extends EventEmitter {
           throw new Error(message);
         }
         this._console.warn(message);
+        // We do NOT want consumers to use a module that is ambiguous.
+        delete moduleMap[platform];
+        if (Object.keys(moduleMap).length === 1) {
+          delete map[id];
+        }
+        let dupsByPl = hasteMap.duplicates[id];
+        if (dupsByPl == null) {
+          dupsByPl = hasteMap.duplicates[id] = (Object.create(null): any);
+        }
+        const dups = (dupsByPl[platform] = (Object.create(null): any));
+        dups[module[H.PATH]] = module[H.TYPE];
+        dups[existingModule[H.PATH]] = existingModule[H.TYPE];
+        return;
+      }
+
+      const dupsByPl = hasteMap.duplicates[id];
+      if (dupsByPl != null) {
+        const dups = dupsByPl[platform];
+        if (dups != null) {
+          dups[module[H.PATH]] = module[H.TYPE];
+        }
+        return;
```

Comment:
I'm not sure I understand the purpose of this block. Can you elaborate on why we need lines 343-349? As I understand it, if an id with a collision has yet another collision, it will overwrite whatever is currently in the duplicates map for that platform. Why?